hi-ml-histopathology/src/histopathology/models/deepmil.py

```diff
@@ -331,17 +331,18 @@ def get_bag_label(labels: Tensor) -> Tensor:
 
     def update_results_with_data_specific_info(self, batch: dict, results: dict) -> None:
         # WARNING: This is a dummy input until we figure out tiles coordinates retrieval in the next iteration.
+        tile_counts = [tiles.shape[0] for tiles in batch[SlideKey.IMAGE]]
```

Comment:
why do we expect a different number of tiles per slides now?

Reply:
For inference, if we want to use all tiles in a slide it might differ from a slide to another. -> The whole purpose of this PR

Reply:
I see! Are some tests capturing this change? (we now want all outputs NOT to assume all slides have the same number of tiles)

Reply:
I am writing a test for the collate function, what else should be tested in your opinion? maybe add a test in test_slides_datamodule?